pynestml/codegeneration/nest_codegenerator.py

```diff
@@ -401,26 +466,13 @@ def generate_neuron_code(self, neuron: ASTNeuron) -> None:
         """
         if not os.path.isdir(FrontendConfiguration.get_target_path()):
             os.makedirs(FrontendConfiguration.get_target_path())
-        self.generate_model_h_file(neuron)
-        self.generate_neuron_cpp_file(neuron)
 
-    def generate_model_h_file(self, neuron: ASTNeuron) -> None:
-        """
-        For a handed over neuron, this method generates the corresponding header file.
-        :param neuron: a single neuron object.
-        """
-        neuron_h_file = self._template_neuron_h_file.render(self.setup_generation_helpers(neuron))
-        with open(str(os.path.join(FrontendConfiguration.get_target_path(), neuron.get_name())) + '.h', 'w+') as f:
-            f.write(str(neuron_h_file))
-
-    def generate_neuron_cpp_file(self, neuron: ASTNeuron) -> None:
-        """
-        For a handed over neuron, this method generates the corresponding implementation file.
-        :param neuron: a single neuron object.
-        """
-        neuron_cpp_file = self._template_neuron_cpp_file.render(self.setup_generation_helpers(neuron))
-        with open(str(os.path.join(FrontendConfiguration.get_target_path(), neuron.get_name())) + '.cpp', 'w+') as f:
-            f.write(str(neuron_cpp_file))
+        for _model_temp in self._model_templates:
+            file_extension = _model_temp.filename.split('.')[-2]
+            _file = _model_temp.render(self.setup_generation_helpers(neuron))
+            with open(str(os.path.join(FrontendConfiguration.get_target_path(),
+                                       neuron.get_name())) + '.' + file_extension, 'w+') as f:
+                f.write(str(_file))
 
     def setup_generation_helpers(self, neuron: ASTNeuron) -> Dict:
```

Comment:
Rename to get_model_namespace().

Reply:
Thanks for this suggestion. Renamed it now.